qemu/tests/qemu_guest_agent.py

```diff
@@ -1310,50 +1311,45 @@ def gagent_check_get_disks(self, test, params, env):
         error_context.context("Check all disks info in a loop.",
                               logging.info)
         disks_info_qga = self.gagent.get_disks()
-        cmd_get_rawdisk_guest = params['get_rawdisk_guest_cmd']
-        raw_disk_guest = session.cmd_output(cmd_get_rawdisk_guest)
-        for disk_info in disks_info_qga:
-            diskname = disk_info['name']
-            if diskname in raw_disk_guest:
-                cmd_rawdisk_guest = params['rawdisk_guest_cmd'] % diskname
-                disk_info_guest = session.cmd_output(cmd_rawdisk_guest)
-            else:
-                cmd_diskinfo_guest = params['diskinfo_guest_cmd'] % diskname
-                disk_info_guest = session.cmd_output(cmd_diskinfo_guest)
-            disk_info_guest = eval(disk_info_guest.strip().strip(','))
-
-            error_context.context("Check disk name", logging.info)
-            if disk_info['name'] != disk_info_guest['kname']:
-                test.fail("Disk %s name is different "
-                          "between guest and qga." % diskname)
-
-            error_context.context("Check dependencies", logging.info)
-            if disk_info_guest['type'] == 'disk':
-                dependencies = 'null'
-                try:
-                    if disk_info['dependencies'][0] != dependencies:
-                        test.error("Disk %s dependencies "
-                                   "should be null." % diskname)
-                except IndexError:
-                    logging.info("Disk '%s' dependencies is null", diskname)
+        cmd_diskinfo_guest = params['diskinfo_guest_cmd']
+        disks_info_guest = session.cmd_output(cmd_diskinfo_guest)
+        disks_info_guest = json.loads(disks_info_guest)['blockdevices']
+
+        for disk_info_guest in disks_info_guest:
+            diskname = disk_info_guest['kname']
+            error_context.context("Check properties of disk %s"
+                                  % diskname, logging.info)
+            for disk_info_qga in disks_info_qga:
+                if diskname == disk_info_qga['name']:
+                    error_context.context("Check dependencies of disk %s"
+                                          % diskname, logging.info)
+                    if disk_info_guest['type'] == 'disk':
+                        if disk_info_qga['dependencies']:
```

Comment:
Just confirm, since here it is a little bit different,  previously `disk_info_qga['dependencies'][0]` is `NULL` is not also accepted, but now `disk_info_qga['dependencies']` must be empty, is that expected for all guest?

Reply:
It is must be '[]' when the type is `disk`, i des not change this logic with this patch.

Reply:
I got your point that we need make it work with both   `[]` and `['null']`.   Will update it to work with both, thanks @vivianQizhu

Reply:
Updated, thanks!